### ISSUE TITLE
error in text output if heading is first element on a page

### ISSUE THREAD

**Issue by manjarofan** [dstosberg/odt2txt#9]
**Explanatory introduction:**
- Obviously Libre/Open Office places elements inside .odt files that effect a page-break when the document is displayed in the GUI-program.
- Obviously the concrete location of such soft page-breaks and therefore the location of the inserted elements can differ for the same document between different office versions.
- When the .odt file is translated to raw XML by using the --raw option of odt2txt, then you can find tags namend `<text:soft-page-break/>` inside the XML stream/file.
- When the output of odt2txt is plain text a heading in the .odt file should be transcoded to

```
    This is a heading!
    ------------------
```

**The bug:**
1. When a heading is the first element on a new page of an .odt document (because the previous page is full), then in the plain text representation generated from this .odt file by odt2txt the underlining for this heading will be missing.
2. When the next element that follows this heading in the .odt file is also a heading, then the line break between these two headings will be missing.

**Expected behaviour:**
Those elements inserted by LibreOffice in order to cause soft page-breaks do not belong to the real content of an .odt document, they are meta content.
->They should not influence the plain text output.
->They should be ignored in XML output, too. Or this should be configurable with an option.

**Affected Versions:**
I did test it for: version 0.5 (compiled from Source), odt2txt-0.4-12.1.2@openSUSE 13.2, odt2txt-0.4-10.1.2@openSUSE 13.1 

**Additional Information:**
For a git repository I configured odt2txt to work as a diff tool for .odt files like it is suggested [here](https://www.thomas-krenn.com/de/wiki/Git_diff_mit_odt-Dateien).
Here's an example for an output off 'git diff' that is incorrect:

```
diff --git a/doc/tuxminds_de/tuxminds_anleitung_3.81.odt b/doc/tuxminds_de/tuxminds_anleitung_3.81.odt
index 09fc4eb..16cef78 100644
--- a/doc/tuxminds_de/tuxminds_anleitung_3.81.odt
+++ b/doc/tuxminds_de/tuxminds_anleitung_3.81.odt
@@ -1,7 +1,7 @@

   TuxMinds

-  Benutzerhandbuch für die Version >= 3.81
+  Benutzerhandbuch für die Version >= 3.96

   Inhaltsverzeichnis

@@ -719,11 +719,13 @@
   nicht)

   Abbruch
-  -------

   siehe Start -nomen est omen-

-  Grundfunktionen der MaustastenLinke Maustaste:
+  Grundfunktionen der Maustasten
+  ------------------------------
+
+  Linke Maustaste:
   ----------------

   Klick auf die Arbeitsfläche+Drag zum Markieren / Umreißen eines

```

Comments to that output:
@@ -1,7 +1,7 @@: I loaded the .odt file into LibreOffice (v4.1.6.2, tried with 5.0, too) changed '3.81' to '3.96' and saved the document in order to test 'odt2txt' in 'git diff'.
So, this difference is O.K. !
@@ -719,11 +719,13 @@: In that version of LibreOffice/OpenOffice, in which the document was modified last (before I did it for testing), the soft page-break had obviously been at another position, in particular in front of the heading `Grundfunktionen der Maustasten`.  And when I executed `File->Save` it has moved before `Abbruch`. Because of the bug `-------` is deleted after "Abbruch" and `------------------------------` is added after  `Grundfunktionen der Maustasten`. 
The line-break, the underlining and an empty line between the two headings `Grundfunktionen der Maustasten` and `Linke Maustaste:` is correct in the new version and incorrect in the old.

Here is the same section from the XML output of the unmodified .odt file:

```
<text:p text:style-name="P10"/>
<text:h text:style-name="Heading_20_2" text:outline-level="2">Abbruch</text:h>
<text:p text:style-name="Standard">
  <text:span text:style-name="T12">siehe Start -nomen est omen-</text:span>
</text:p>
<text:h text:style-name="Heading_20_2" text:outline-level="2"><text:soft-page-break/>Grundfunktionen der Maustasten</text:h>
<text:h text:style-name="Heading_20_2" text:outline-level="2">Linke Maustaste:</text:h>
```

**Comment by dstosberg** [comment]
Hi, can you provide (link to) a sample file that triggers this behaviour?

**Comment by manjarofan** [comment]
On 02.05.2016 20:07, dstosberg wrote:
>
> Hi, can you provide (link to) a sample file that triggers this behaviour?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/dstosberg/odt2txt/issues/9#issuecomment-216313335>
>
Hi,

yes, I'll try to directly send them to you as attachments.

But I'm new at github and I don't know if it will work.

So, please give me a short feedback, even if you'll successfully receive 
those files.
Thanks!

The error occurs at the top of
<Page>: <Heading>
5: 4 Installation und Start
6: 5 Zugriff auf die Roboter über Linux Devices
8: Vorbemerkungen
10: Der Entwurfsmodus
12: Abbruch
13: 11 Skins (unchanged (=orig) file only)

Winfried

P.S. Another point seems to be (just saw it), that headings with a 
numbering loose this numbering in text mode.
But this is not so problematic when used with diff, because it's consistent.

**Comment by manjarofan** [comment]
(I got an error message when trying to upload a file here. Personal mailing seems impossible at github?)
So I suggest you to extract the file out of an archive:
Here's the [link to the project](https://sourceforge.net/projects/tuxminds/files/testing/).
There you'll find the file 'doc/tuxminds_de/tuxminds_anleitung_3.81.odt' in the archive 'src_tuxminds_3.96_20141203.tbz'.

The bug occurs at the top of
<Page>: <Heading>
5: 4 Installation und Start
6: 5 Zugriff auf die Roboter über Linux Devices
8: Vorbemerkungen
10: Der Entwurfsmodus
12: Abbruch
13: 11 Skins (unchanged (=orig) file only) 

I looked at the whole text fiIe generated from 'src_tuxminds_3.96_20141203.tbz'
and noticed that the bug occurs in at least one other constellations, too: 
- When a text:list ends just in front of a heading (</text:list> is the name of the closing XML tag when the output type is raw XML):
'Grundlegende Bedienung' and 'Betrieb ohne Oberfläche (Batchbetrieb)' (that follows immediately) are both headings (at page 7 of the .odt file) and they are *not* at the top of that page but the bug occurs in the generated txt file.
